spec/system/user_management_spec.rb

```diff
@@ -3,77 +3,70 @@
 require 'rails_helper'
 
 RSpec.describe 'User Management' do
+  let!(:user) { create(:user) }
+
   before do
-    @user = create(:user)
     when_current_user_is :admin
+    visit users_path
   end
 
-  context 'deleting a user unsuccessfully', :js do
-    before do
-      @log = create(:log_entry, user: @user)
+  context 'when deleting a user', :js do
+    subject :submit do
+      page.accept_confirm 'Are you sure?' do
+        within('tr', text: user.name) { click_on 'Destroy' }
+      end
     end
 
-    it 'puts the errors in the flash' do
-      visit users_path
-      # Click the 'destroy' link in the user's row
-      page.accept_confirm 'Are you sure?' do
-        within('tr', text: @user.name) { click_on 'Destroy' }
+    context 'when a user is not deletable' do
+      before { create(:log_entry, user:) }
+
+      it 'puts the errors in the flash' do
+        submit
+        expect(page).to have_text('Cannot delete record because dependent log entries exist')
       end
-      expect(page).to have_text(
-        'Cannot delete record because dependent log entries exist'
-      )
     end
-  end
 
-  context 'deleting a user successfully', :js do
-    it 'deletes the user and says it did' do
-      visit users_path
-      page.accept_confirm 'Are you sure?' do
-        within('tr', text: @user.name) { click_on 'Destroy' }
+    context 'when a user is deletable' do
+      it 'deletes the user and says it did' do
```

Comment:
I don't see a test for whether it actually deletes the user.... then again later it's got the same kind of implications. Is that just because it's a system test, don't bother testing that it actually updates that's for request/model specs?
```suggestion
      it 'says it deletes the user' do
```

Reply:
Same for all of these examples, actually.

Reply:
> Same for all of these examples, actually.

Yeah i noticed that after and that's when I started thinking I might be going a little too hard considering the state of these specs